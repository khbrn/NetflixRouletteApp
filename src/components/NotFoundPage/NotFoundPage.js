import React from "react";
import { Link } from "react-router-dom";

import "./NotFoundPage.css";

const NotFoundPage = () => {
  return (
    <div className="error-message">
      <h1>Page Not Found</h1>
      <h2>404</h2>
      <button type="button">
        <Link to="/">BACK TO HOME PAGE</Link>
      </button>
    </div>
  );
};

export default NotFoundPage;
