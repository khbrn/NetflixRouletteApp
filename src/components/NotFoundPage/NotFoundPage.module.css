.error-message {
  margin: 16px;
}

.error-message h1,
.error-message h2 {
  margin-bottom: 5px;
}
