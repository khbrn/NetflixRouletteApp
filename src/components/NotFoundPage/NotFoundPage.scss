.error-message {
  margin: 16px;

  h1,
  h2 {
    margin-bottom: 5px;
  }
}
