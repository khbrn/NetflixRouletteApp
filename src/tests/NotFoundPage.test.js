import React from "react";
import { MemoryRouter } from "react-router-dom";
import { unmountComponentAtNode } from "react-dom";
import { screen, render } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { createMemoryHistory } from "history";
import pretty from "pretty";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

import NotFoundPage from "../components/NotFoundPage/NotFoundPage";

describe("NotFoundPage component", () => {
  it("shows the not found page message for incorrect URL address", () => {
    const history = createMemoryHistory();
    history.push("/movie-not-found");

    const tree = renderer
      .create(
        <MemoryRouter location={history.location} navigator={history}>
          <NotFoundPage />
        </MemoryRouter>
      )
      .toJSON();

    render(
      <MemoryRouter location={history.location} navigator={history}>
        <NotFoundPage />
      </MemoryRouter>
    );

    expect(screen.getByText(/page not found/i)).toBeInTheDocument();
    expect(screen.getByText(/404/i)).toBeInTheDocument();
    expect(screen.getByText(/back to home page/i)).toBeInTheDocument();

    expect(tree).toMatchSnapshot();
  });
});
