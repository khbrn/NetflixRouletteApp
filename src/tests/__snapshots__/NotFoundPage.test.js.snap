// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component shows the not found page message for incorrect URL address 1`] = `
<div
  className="error-message"
>
  <h1>
    Page Not Found
  </h1>
  <h2>
    404
  </h2>
  <button
    type="button"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      BACK TO HOME PAGE
    </a>
  </button>
</div>
`;
